examples/threading/threading.c

```diff
@@ -0,0 +1,86 @@
+#include "threading.h"
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+// Optional: use these functions to add debug or error prints to your application
+#define DEBUG_LOG(msg,...)
+//#define DEBUG_LOG(msg,...) printf("threading: " msg "\n" , ##__VA_ARGS__)
+#define ERROR_LOG(msg,...) printf("threading ERROR: " msg "\n" , ##__VA_ARGS__)
+
+void* threadfunc(void* thread_param)
+{
+
+    // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
+    // hint: use a cast like the one below to obtain thread arguments from your parameter
+    //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    
+    struct thread_data *th_data = (struct thread_data *)thread_param;
+
+    // Sleep for wait_to_obtain_ms milliseconds
+    if (usleep(th_data->wait_to_obtain_ms * 1000) != 0) {
+    th_data->thread_complete_success = false;
+    ERROR_LOG("Failed while waiting to obtain the mutex");
+    }
+
+    // Obtain the mutex
+    if (pthread_mutex_lock(th_data->mutex) != 0) {
+    th_data->thread_complete_success = false;
+    ERROR_LOG("Failed to lock the mutex");
+    }
+
+    // Sleep for wait_to_release_ms milliseconds
+    if (usleep(th_data->wait_to_release_ms * 1000) != 0) {
+    th_data->thread_complete_success = false;
+    ERROR_LOG("Failed while waiting to release the mutex");
+    }
+
+    // Release the mutex
+    if (pthread_mutex_unlock(th_data->mutex) != 0) {
+    th_data->thread_complete_success = false;
+    ERROR_LOG("Failed to unlock the mutex");
+    }
+
+    // Set thread complete success flag
+    th_data->thread_complete_success = true;
```

Comment:
@Suhas-Reddy-S what would happen if the `usleep` failed above and set `thread_complete_success` to false?